Source/DafnyCore/AST/Statements/AssignSuchThatStmt.cs

```diff
@@ -76,6 +76,10 @@ public AssignSuchThatStmt(RangeToken rangeToken, List<Expression> lhss, Expressi
     base.Resolve(resolver, resolutionContext);
 
     if (AssumeToken != null) {
+      if (!AssumeToken.IsExplicitAxiom()) {
+        resolver.Reporter.Warning(MessageSource.Resolver, ErrorDetail.ErrorID.None, AssumeToken.Token, "Assume statement has no {:axiom} annotation.");
```

Comment:
It might be an improvement for this message to mention the context more explicitly, i.e. that it's in an assign-such-that statement.